superset/views/core.py

```diff
@@ -1869,6 +1875,19 @@ def dashboard(  # pylint: disable=too-many-locals
             if key not in [param.value for param in utils.ReservedUrlParameters]
         }
 
+        extra_jwt = {}
+        if feature_flag_manager.is_feature_enabled("DASHBOARD_RBAC"):
```

Comment:
There is a project underway to make Dashboards part of a single page app. This code will only work if dashboards are accessed from this route, but in a SPA that will not necessarily be the case. If you click a link to a dashboard from the dashboards list or the homepage, the routing will be done on the frontend. Maybe there should be a separate endpoint that the frontend calls to get the jwt.

Reply:
I'm also not really clear on why the jwt is needed. Some more background info would be helpful.

Reply:
Ok so the jwt is being used to manage read data access of charts that exists on a dashboard implicitly just by having a access to the dashboard

This really upgrade the dashboard access RBAC mechanism by also giving Temporary access to datasets that are being used within the dash lard without the need to explicitly giving permission to those datasets

Reply:
I'm totally aware of SPA project and I plan to also support and implement the dashboard jwt feature under that as well

But for now until SPA maturity I want to to be supported in the old style

Reply:
I'm expecting to have a PR up that removes dashboard as a standalone app this week, so I think compatibility is an important consideration now.

What about adding logic to the security of the chart data and dataset endpoints, to check whether access should be granted at the time of request? Is that an option? Sorry if I'm missing something, I'd just like to fully understand the need for this JWT pattern. Is the time limit for access a feature requirement or an implementation detail?

Reply:
I'm a little bit wary of the security implication of allowing dashboard readers to have access to charts and datasets they explicitly don't have access to.

What if someone creates a dashboard with charts that use datasets they don't have access to, but add themselves to the allowed list of roles? What is the access control flow look like? Would you restrict who can change dashboard roles only to those who have admin access to the underlying datasets? Because if someone has view access to a dataset or edit access to a dashboard, it doesn't mean they can just publish this dataset or the content of this dashboard to any audience.